progress-bar-ui.js

```diff
@@ -0,0 +1,337 @@
+define([
+    '$',
+    'deckard'
+], function($) {
+
+    // Utilities
+
+    var _clampPercentage = function(percentage) {
+        if (percentage > 1) {
+            percentage = 1;
+        } else if (percentage < 0) {
+            percentage = 0;
+        }
+
+        return percentage;
+    };
+
+    var _uuid = (function() {
+        var counter = 0;
```

Comment:
I’m not the best JS dev in the world – does this counter increment safely for all instances, ensuring each instance gets unique IDs?